File: 0559-maximum-depth-of-n-ary-tree/0559-maximum-depth-of-n-ary-tree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root, int depth = 0) {
        if(root == nullptr) return 0;
        for(const auto &child : root->children) 
            depth = max(depth, maxDepth(child));
        return 1 + depth;
    }
};

File: 0559-maximum-depth-of-n-ary-tree/0559-maximum-depth-of-n-ary-tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node', depth: int = 0) -> int:
        if root is None: return 0
        for child in root.children:
            depth = max(depth, self.maxDepth(child))
        return 1 + depth